File: app/src/main/java/com/filemanager/fileexplorer/myfile/cleanmaster/free/Iinterface/CallbackUpdateMusic.java
package com.filemanager.fileexplorer.myfile.cleanmaster.free.Iinterface;


public class CallbackUpdateMusic {
    private static CallbackUpdateMusic callbackupdate;
    private OncustomStateListen stateListen;


    public interface OncustomStateListen {
        void statechange();
    }

    private CallbackUpdateMusic() {
    }

    public static CallbackUpdateMusic getInstance() {
        if (callbackupdate == null) {
            callbackupdate = new CallbackUpdateMusic();
        }
        return callbackupdate;
    }

    public void setStateListen(OncustomStateListen oncustomStateListen) {
        this.stateListen = oncustomStateListen;
    }

    public void change() {
        notidatachange();
    }

    protected void notidatachange() {
        this.stateListen.statechange();
    }
}

File: app/src/main/java/com/demo/filemanager/Iinterface/CallbackUpdateMusic.kt
package com.demo.filemanager.Iinterface

class CallbackUpdateMusic private constructor() {
    private var stateListen: OncustomStateListen? = null

    interface OncustomStateListen {
        fun statechange()
    }

    fun setStateListen(oncustomStateListen: OncustomStateListen?) {
        stateListen = oncustomStateListen
    }

    fun change() {
        notidatachange()
    }

    protected fun notidatachange() {
        stateListen!!.statechange()
    }

    companion object {
        private var callbackupdate: CallbackUpdateMusic? = null
        val instance: CallbackUpdateMusic?
            get() {
                if (callbackupdate == null) {
                    callbackupdate = CallbackUpdateMusic()
                }
                return callbackupdate
            }
    }
}
